src/Tracing/GuzzleTracingMiddleware.php

```diff
@@ -13,27 +17,58 @@
  */
 final class GuzzleTracingMiddleware
 {
-    public static function trace(?HubInterface $hub = null): \Closure
+    public static function trace(?HubInterface $hub = null): Closure
     {
-        return function (callable $handler) use ($hub): \Closure {
-            return function (RequestInterface $request, array $options) use ($hub, $handler) {
+        return static function (callable $handler) use ($hub): Closure {
+            return static function (RequestInterface $request, array $options) use ($hub, $handler) {
                 $hub = $hub ?? SentrySdk::getCurrentHub();
                 $span = $hub->getSpan();
-                $childSpan = null;
 
-                if (null !== $span) {
-                    $spanContext = new SpanContext();
-                    $spanContext->setOp('http.guzzle');
-                    $spanContext->setDescription($request->getMethod() . ' ' . $request->getUri());
-
-                    $childSpan = $span->startChild($spanContext);
+                if (null === $span) {
+                    return $handler($request, $options);
                 }
 
-                $handlerPromiseCallback = static function ($responseOrException) use ($childSpan) {
-                    if (null !== $childSpan) {
-                        $childSpan->finish();
+                $spanContext = new SpanContext();
+                $spanContext->setOp('http.client');
+                $spanContext->setDescription($request->getMethod() . ' ' . $request->getUri());
+
+                $childSpan = $span->startChild($spanContext);
+
+                $request->withHeader('sentry-trace', $childSpan->toTraceparent());
+
+                $handlerPromiseCallback = static function ($responseOrException) use ($hub, $request, $childSpan) {
+                    $childSpan->finish();
+
+                    $response = $responseOrException instanceof ResponseInterface ? $responseOrException : null;
```

Comment:
Personally I dislike a `if/elif/else` block if we don't have to.

The current one is the expected case and the `GuzzleRequestException` is a special handling of the value, and as an added bonus the psalm suppression also looks a lot better without being nested inside and `if/elif/else` block.

Happy to change it if needed but I think it's pretty clear like this.

Reply:
>I think it's pretty clear like this

I suggested the change because I think the opposite, when I read the code I had to understand the ternary operator first and then another if statement before knowing what `$response` contains. I think that `elseif` would make the code more straightforward and readable on first sight

Reply:
I'm not going to argue about this, potato potatoe. 

It's fine I already said I would be glad to change it so I will you just had to say it :)